pkg/coord/adapter.go

```diff
@@ -65,8 +64,28 @@ func (a *adapter) ListAllKeyRanges(ctx context.Context) ([]*kr.KeyRange, error)
 }
 
 func (a *adapter) AddKeyRange(ctx context.Context, kr *kr.KeyRange) error {
+	_, err := a.GetShardInfo(ctx, kr.ShardID)
+	if err != nil {
+		return err
+	}
+
+	dataspaces, err := a.ListDataspace(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
why do we execute this checks here if we have them in `AddKeyRangeWithChecks`?

Reply:
Because we don't have QDB struct in adapter